IO-pc-local/src/app/forms/creator/unsaved-form.service.ts

```diff
@@ -0,0 +1,46 @@
+import {Injectable} from '@angular/core';
+import {FormField, FormFieldId, FormType} from "../model";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UnsavedFormService {
+  private firstFreeFieldId = 0
+  private currentForm: FormType = []
+
+  get form(): FormType {
+    return [...this.currentForm]
```

Comment:
Kopiowanie za pomocą destrukturyzacji jest płytkie, więc elementy nowej tablicy to dokładnie te same referencje co elementy oryginalnej tablicy. Czy chcemy je udostępniać i pozwalać na bezpośrednią ich modyfikację?

Reply:
Bardzo dobra uwaga, aczkolwiek w tym przypadku z racji na `readonly` w interfejsie wydaje mi się, że jesteśmy bezpieczni: [Playground Link](https://www.typescriptlang.org/play/?#code/C4TwDgpgBAYg9gJwLYwJYQDYBMCSWoC8UAdgK5IBGECAUDascNQGYCGAxtPMmpvgN40oUBBFZY4xDCCiosALliIU6bHiEixEqTICOpCAGdgqSYuMIGAcxoBfOqEhLkAFXDQi3FXwDaAXTp2SWModlIEUUYvRS83JyIfDWF+OUUAWgBWABoofSMTMygAIgBpVCQoUCgAdyg4KwBnjGgAK3yIQG1AUIAvQCFAOGr0AGtSAH4i2yykqBSFKDSAFhy841NiRSKAEXJiGSruqEN0CnQoNuNOnv7BiBHxyenZ9IBmZYNVwqKAdXRDA6H6nBKjI4BQEKxAD6ArD+9ymARoQWIISCYHQ+ASADosWEIhAosp4Si0T4AIx+DErArEQjFCB0iBFGhAA)

Jeśli się mylę, wskaż proszę w jaki sposób możnaby to zepsuć

Reply:
Faktycznie, zapomniałem, że w interfejsie pola są ustawione na readonly